cranelift-simplejit/src/memory.rs

```diff
@@ -209,9 +227,20 @@ impl Memory {
             }
         }
     }
+
+    /// Frees all allocated memory regions that would be leaked otherwise.
+    /// Likely to invalidate existing function pointers, causing unsafety.
+    pub unsafe fn free_memory(&mut self) {
+        self.allocations.clear();
+    }
 }
 
-// TODO: Implement Drop to unprotect and deallocate the memory?
+impl Drop for Memory {
+    fn drop(&mut self) {
+        // leak memory to guarantee validity of function pointers
+        Box::leak(self.allocations.split_off(0).into_boxed_slice());
+    }
```

Comment:
Maybe `mem::replace(&mut self.collections, Vec::new()).into_iter().for_each(mem::forget)`? That would prevent leaking the vec itself. It would also prevent having to move all `PtrLen` to a new vec and then a boxed slice before leaking.